pkg/leech/ingest_logs.go

```diff
@@ -180,7 +179,7 @@ func (f *IngestLogsFn) ProcessElement(ctx context.Context, event EventRecord) Le
 // handleMessage is the main event processor. It generates a GitHub token, reads the workflow
 // log files if they exist and persists them to Cloud Storage.
 func (f *IngestLogsFn) handleMessage(ctx context.Context, repoName, ghLogsURL, gcsPath string) error {
-	token, err := f.repoAccessToken(ctx, repoName)
+	token, err := auth.ReadAccessTokenForRepos(ctx, f.ghApp, repoName)
```

Comment:
Curious on thoughts for this, as I will be doing something similar. Does it make sense to just ask for read access to all repos and cache that in the local host?

In theory, Leech will be grabbing logs for an entire organization, so may different repositories may be accessed by a particular instance, so does it make sense to have O(repositories*runners) calls to grab granular tokens, or should we just grab a read all token once in each runner and use that for all calls.

@bradegler Might have opinions here?

Reply:
This is essentially what I'm doing for the review pipeline. Since I need access to all repositories I simply use the GitHub App to fetch a token for all repos when pipeline starts up and just use that token throughout the pipeline.
https://github.com/abcxyz/github-metrics-aggregator/pull/178/files#diff-5b439c7a9f8f1115f2d8c05ba6b558ab32ffcde32aa55d3b18b746a64dd98bb7R105

Reply:
So the token is passed along the pipeline then? For some reason I was thinking we would need to re-generate it at each step.

Reply:
Yes, It gets stored in this config:
https://github.com/abcxyz/github-metrics-aggregator/blob/70753c04547ae797f337c82de9fdd24f059a5da9/pipeline/review/main.go#L122

Which is then used given to the beam DoFn structs:
https://github.com/abcxyz/github-metrics-aggregator/blob/70753c04547ae797f337c82de9fdd24f059a5da9/pkg/review/commit_review_status.go#L189

https://github.com/abcxyz/github-metrics-aggregator/blob/70753c04547ae797f337c82de9fdd24f059a5da9/pkg/review/commit_review_status.go#L204